tests/models/test_dag.py

```diff
@@ -387,243 +388,93 @@ def test_get_num_task_instances(self):
-                  start_date=DEFAULT_DATE)
-
-        with dag:
-            task = DummyOperator(task_id='op1')
-
-        self.assertEqual(
-            task.render_template('', datetime.date(2018, 12, 6), {'foo': 'bar'}),
-            datetime.date(2018, 12, 6)
+            expected.var1, actual.var1, msg="Named tuples may not have been preserved in rendering"
         )
-
-    def test_render_template_datetime_field(self):
-        """Tests if render_template from a datetime field works"""
-
-        dag = DAG('test-dag',
-                  start_date=DEFAULT_DATE)
-
-        with dag:
-            task = DummyOperator(task_id='op1')
-
         self.assertEqual(
-            task.render_template('', datetime.datetime(2018, 12, 6, 10, 55), {'foo': 'bar'}),
-            datetime.datetime(2018, 12, 6, 10, 55)
+            expected.var2, actual.var2, msg="Named tuples may not have been preserved in rendering"
         )
 
     def test_render_template_UUID_field(self):
         """Tests if render_template from a UUID field works"""
-
-        dag = DAG('test-dag',
-                  start_date=DEFAULT_DATE)
-
-        with dag:
+        with DAG('test-dag', start_date=DEFAULT_DATE):
             task = DummyOperator(task_id='op1')
 
         random_uuid = uuid.uuid4()
-        self.assertIs(
-            task.render_template('', random_uuid, {'foo': 'bar'}),
-            random_uuid
-        )
+        self.assertIs(task.render_template(random_uuid, {'foo': 'bar'}), random_uuid)
 
     def test_render_template_object_field(self):
         """Tests if render_template from an object field works"""
-
-        dag = DAG('test-dag',
-                  start_date=DEFAULT_DATE)
-
-        with dag:
+        with DAG('test-dag', start_date=DEFAULT_DATE):
             task = DummyOperator(task_id='op1')
 
         test_object = object()
-        self.assertIs(
-            task.render_template('', test_object, {'foo': 'bar'}),
-            test_object
-        )
-
-    def test_render_template_field_macro(self):
-        """ Tests if render_template from a field works,
-            if a custom filter was defined"""
-
-        dag = DAG('test-dag',
-                  start_date=DEFAULT_DATE,
-                  user_defined_macros=dict(foo='bar'))
-
-        with dag:
-            task = DummyOperator(task_id='op1')
-
-        result = task.render_template('', '{{ foo }}', dict())
-        self.assertEqual(result, 'bar')
-
-    def test_render_template_numeric_field(self):
-        """ Tests if render_template from a field works,
-            if a custom filter was defined"""
-
-        dag = DAG('test-dag',
-                  start_date=DEFAULT_DATE,
-                  user_defined_macros=dict(foo='bar'))
-
-        with dag:
-            task = DummyOperator(task_id='op1')
-
-        result = task.render_template('', 1, dict())
-        self.assertEqual(result, 1)
+        self.assertIs(task.render_template(test_object, {'foo': 'bar'}), test_object)
```

Comment:
Here too?

Reply:
Same here, `assertIs` is being used here instead of `assertEqual` like in the general parameterized test.

Reply:
Is `assertIs` an important distinction here?

Reply:
Just double checked, it's not necessary (`assertEqual` also works) but does have the same problem as with the uuid test: we need to initialize a variable before the test, which is different from the other parameterized tests.

I've thought about how to remove more lines of code and somehow parameterize the uuid, object and namedtuple tests. The uuid & object tests have different behaviour from the other parameterized tests and thus I wouldn't hack them together (they verify if the input object is the same as before & after templating). We could merge the two thought:

```python
@parameterized.expand([
    (object(),),
    (uuid.uuid4(),)
])
def test_render_template_fields_no_change(self, test_input):
    """Tests if the templated object remains unchanged"""
    with DAG('test-dag', start_date=DEFAULT_DATE):
        task = DummyOperator(task_id='op1')

    result = task.render_template(test_input, {'foo': 'bar'})
    self.assertEqual(test_input, result)
```

The namedtuple test could be merged together with the first big parameterized test. Although there is a way to inline the declaration, that would become real ugly IMO, so it's better to simply declare the namedtuple class at the top.